ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java

```diff
@@ -1181,6 +1212,55 @@ public void afterCompletion(final int status) {
         }
     }
 
+    private class TaskPostPersist extends java.util.TimerTask {
+        private final TimerImpl timer;
+        private long delta = 0;
+
+        TaskPostPersist(TimerImpl timer) {
+            this.timer = timer;
+        }
+
+        TaskPostPersist(TimerImpl timer, long delta) {
+            this.timer = timer;
+            this.delta = delta;
+        }
+
+        @Override
+        public void run() {
+            final ExecutorService executor = executorServiceInjectedValue.getOptionalValue();
+            if (executor != null) {
```

Comment:
This will never be null, as this is not an optional dependency. Functionally it makes no difference but it could be confusing to someone reading the code.

Reply:
I'm sorry, but the method name is "getOptionalValue", so I am already confused. BTW, I made this check the same way as in the main "Task" class below (line 1282).

Reply:
You should use getValue() rather than getOptionalValue() (this is not a big deal, but just makes the code a bit clearer)

Reply:
Yes, I've made this change in the last commit. But there is some problem in GITHUB, again. :(
Last time, I closed my pull request and opened a new one with the full patch. Maybe, I have to do it again.